app/services/survey/withdraw.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module Survey
+  class Withdraw
+    include ServicePattern
+
+    def initialize(trainee:)
+      @trainee = trainee
+    end
+
+    def call
+      create_contact_in_mailing_list_body = create_contact_in_mailing_list_response.body
+      contact_lookup_id = JSON.parse(create_contact_in_mailing_list_body)["result"]["contactLookupId"]
+      create_distribution_response(contact_lookup_id)
+    end
+
+  private
+
+    attr_reader :trainee
+
+    delegate :first_names, :last_name, :email, :withdraw_date, :training_route, to: :trainee
+
+    def create_contact_in_mailing_list_response
+      @create_contact_in_mailing_list_response ||= QualtricsApi::Client::Request.post("/directories/#{directory_id}/mailinglists/#{mailing_list_id}/contacts", body: create_contact_in_mailing_list_body)
+    end
+
+    def create_contact_in_mailing_list_body
```

Comment:
Reference https://api.qualtrics.com/fc43c25b3f3c3-create-contact-in-mailing-list#request-body